pkg/xds/envoy/listeners/listener_builder.go

```diff
@@ -68,5 +68,15 @@ func (c *ListenerBuilderConfig) AddV3(configurer v3.ListenerConfigurer) {
 type ListenerBuilderOptFunc func(config *ListenerBuilderConfig)
 
 func (f ListenerBuilderOptFunc) ApplyTo(config *ListenerBuilderConfig) {
-	f(config)
+	if f != nil {
```

Comment:
That's kind of a developer error if `f` is equal to `nil` here, it could be some misconfiguration of `Builders` for example. Isn't it better to immediately panic rather than silently do something wrong?

Reply:
We use this in some of the configures, which are conditional. It allows the test for whether to be apply the conditioner to be make outside the callback. We can just return a nil pointer.